katz/src/main/kotlin/katz/data/OptionT.kt

```diff
@@ -24,65 +24,69 @@ typealias OptionTF<F> = HK<OptionT.F, F>
-            OptionT(F.map(value, { it.map(f) }))
+    inline fun <B> map(crossinline f: (A) -> B): OptionT<F, B> =
+            OptionT(MF, MF.map(value, { it.map(f) }))
 
-    fun getOrElse(F: Functor<F>, default: () -> A): HK<F, A> = F.map(value, { it.getOrElse(default) })
+    fun getOrElse(default: () -> A): HK<F, A> = MF.map(value, { it.getOrElse(default) })
 
-    inline fun getOrElseF(F: Monad<F>, crossinline default: () -> HK<F, A>): HK<F, A> = F.flatMap(value, { it.fold(default, { F.pure(it) }) })
+    inline fun getOrElseF(crossinline default: () -> HK<F, A>): HK<F, A> = MF.flatMap(value, { it.fold(default, { MF.pure(it) }) })
 
-    inline fun filter(F: Functor<F>, crossinline p: (A) -> Boolean): OptionT<F, A> = OptionT(F.map(value, { it.filter(p) }))
+    inline fun filter(crossinline p: (A) -> Boolean): OptionT<F, A> = OptionT(MF, MF.map(value, { it.filter(p) }))
 
-    inline fun forall(F: Functor<F>, crossinline p: (A) -> Boolean): HK<F, Boolean> = F.map(value, { it.forall(p) })
+    inline fun forall(crossinline p: (A) -> Boolean): HK<F, Boolean> = MF.map(value, { it.forall(p) })
 
-    fun isDefined(F: Functor<F>): HK<F, Boolean> = F.map(value, { it.isDefined })
+    fun isDefined(): HK<F, Boolean> = MF.map(value, { it.isDefined })
 
-    fun isEmpty(F: Functor<F>): HK<F, Boolean> = F.map(value, { it.isEmpty })
+    fun isEmpty(): HK<F, Boolean> = MF.map(value, { it.isEmpty })
 
-    inline fun orElse(F: Monad<F>, crossinline default: () -> OptionT<F, A>): OptionT<F, A> =
-            orElseF(F, { default().value })
+    inline fun orElse(crossinline default: () -> OptionT<F, A>): OptionT<F, A> =
+            orElseF({ default().value })
 
-    inline fun orElseF(F: Monad<F>, crossinline default: () -> HK<F, Option<A>>): OptionT<F, A> =
-            OptionT(F.flatMap(value) {
+    inline fun orElseF(crossinline default: () -> HK<F, Option<A>>): OptionT<F, A> =
+            OptionT(MF, MF.flatMap(value) {
                 when (it) {
-                    is Option.Some<A> -> F.pure(it)
+                    is Option.Some<A> -> MF.pure(it)
                     is Option.None -> default()
                 }
             })
 
-    inline fun <B> transform(F: Monad<F>, crossinline f: (Option<A>) -> Option<B>): OptionT<F, B> =
-            OptionT(F.map(value, { f(it) }))
+    inline fun <B> transform(crossinline f: (Option<A>) -> Option<B>): OptionT<F, B> =
+            OptionT(MF, MF.map(value, { f(it) }))
 
-    inline fun <B> subflatMap(F: Monad<F>, crossinline f: (A) -> Option<B>): OptionT<F, B> =
-            transform(F, { it.flatMap(f) })
+    inline fun <B> subflatMap(crossinline f: (A) -> Option<B>): OptionT<F, B> =
+            transform({ it.flatMap(f) })
 
     //TODO: add toRight() and toLeft() once EitherT it's available
-}
\ No newline at end of file
+}
+
+
+inline fun <reified F, A> optionT(value: HK<F, Option<A>>, MF : Monad<F> = monad<F>()): OptionT<F, A> = OptionT(MF, value)
```

Comment:
why not use the same magic from `Kleisli` invoke operator here instead of this function?